src/coreclr/src/tools/aot/ILCompiler.Compiler/IL/Stubs/StartupCode/StartupCodeMainMethod.cs

```diff
@@ -92,6 +92,22 @@ public override MethodIL EmitIL()
                 codeStream.Emit(ILOpcode.call, emitter.NewToken(initEntryAssembly));
             }
 
+            // Initialize COM apartment
+            MethodDesc initApartmentState = startup.GetMethod("InitializeApartmentState", null);
+            if (initApartmentState != null)
+            {
+                if (_mainMethod.HasCustomAttribute("System", "STAThreadAttribute"))
+                {
+                    codeStream.EmitLdc(0 /* ApartmentState.STA */);
```

Comment:
Nit: We could probably just use the enum directly. The enums are guaranteed to not go out of sync.